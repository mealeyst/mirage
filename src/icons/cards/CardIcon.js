import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import Amex from './Amex.base'
import Default from './Default.base'
import Diners from './Diners.base'
import Discover from './Discover.base'
import JCB from './JCB.base'
import Mastercard from './Mastercard.base'
import Maestro from './Maestro.base'
import Paypal from './Paypal.base'
import Visa  from './Visa.base'

const CardIconBase = ({ brand, className }) => {
  let Child = null
    switch (brand) {
      case 'amex':
        Child = Amex
        break
      case 'diners':
        Child = Diners
        break
      case 'discover':
        Child = Discover
        break
      case 'jcb':
        Child = JCB
        break
      case 'mastercard':
        Child = Mastercard
        break
      case 'maestro':
        Child = Maestro
        break
      case 'paypal':
        Child = Paypal
        break
      case 'visa':
        Child = Visa
        break
      default:
        Child = Default
  }
  return (
    <Child className={className} />
  )
}

const CardIcon = styled(CardIconBase)`
  fill: ${props => props.theme.colors.navy};
`

CardIcon.propTypes = {
  brand: PropTypes.string.isRequired,
  theme: PropTypes.shape({
    colors: PropTypes.shape({
      rocketBlue: PropTypes.string
    })
  }).isRequired
}

/** @component */
export default CardIcon
