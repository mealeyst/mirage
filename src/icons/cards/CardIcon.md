```js
<CardIcon brand='amex' />

<CardIcon brand='diners' />

<CardIcon brand='discover' />

<CardIcon brand='jcb' />

<CardIcon brand='mastercard' />

<CardIcon brand='maestro' />

<CardIcon brand='paypal' />

<CardIcon brand='visa' />
```
