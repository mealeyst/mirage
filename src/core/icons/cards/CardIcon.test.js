import React from 'react'
import 'jest-styled-components'

import { CardIcon } from 'SRC'
import { CardIconBase } from './CardIcon'

import Amex from './Amex.base'
import Default from './Default.base'
import Diners from './Diners.base'
import Discover from './Discover.base'
import JCB from './JCB.base'
import Mastercard from './Mastercard.base'
import Maestro from './Maestro.base'
import Paypal from './Paypal.base'
import Visa  from './Visa.base'

const { mountWithTheme } = global

describe('(Styled Component) CardIcon', () => {
  const createCardIcon = (props) => {
    let brand = ''
    if (props && props.brand) {
      brand = props.brand
    }
    return mountWithTheme(<CardIcon {...props} brand={brand} />)
  }

  test('matching the snapshot', () => {
    expect(createCardIcon())
    .toMatchSnapshot()
  })
})


describe('(Component) CardIconBase', () => {
  const createCardIconBase = (props) => {
    let brand = ''
    if (props && props.brand) {
      brand = props.brand
    }
    return mountWithTheme(<CardIconBase {...props} brand={brand} />)
  }

  test('setting the className', () => {
    const className = 'example-class'
    expect(
      createCardIconBase({className: className}).prop('className')
    ).toEqual(className)
  })

  test('setting the brand to amex', () => {
    expect(createCardIconBase({brand: 'amex'}).find(Amex).length)
    .toEqual(1)
  })

  test('setting the brand to diners', () => {
    expect(createCardIconBase({brand: 'diners'}).find(Diners).length)
    .toEqual(1)
  })

  test('setting the brand to discover', () => {
    expect(createCardIconBase({brand: 'discover'}).find(Discover).length)
    .toEqual(1)
  })

  test('setting the brand to jcb', () => {
    expect(createCardIconBase({brand: 'jcb'}).find(JCB).length)
    .toEqual(1)
  })

  test('setting the brand to mastercard', () => {
    expect(createCardIconBase({brand: 'mastercard'}).find(Mastercard).length)
    .toEqual(1)
  })

  test('setting the brand to maestro', () => {
    expect(createCardIconBase({brand: 'maestro'}).find(Maestro).length)
    .toEqual(1)
  })

  test('setting the brand to paypal', () => {
    expect(createCardIconBase({brand: 'paypal'}).find(Paypal).length)
    .toEqual(1)
  })

  test('setting the brand to visa', () => {
    expect(createCardIconBase({brand: 'visa'}).find(Visa).length)
    .toEqual(1)
  })

  test('not setting the brand', () => {
    expect(createCardIconBase({brand: ''}).find(Default).length)
    .toEqual(1)
  })
})
