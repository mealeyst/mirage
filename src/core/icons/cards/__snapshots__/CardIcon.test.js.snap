// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Styled Component) CardIcon matching the snapshot 1`] = `
.c0 {
  fill: #00003C;
}

<Styled(CardIconBase)
  brand=""
>
  <CardIconBase
    brand=""
    className="c0"
  >
    <Default
      className="c0"
    >
      <BaseCardIcon
        box="0 0 750 472"
        className="c0"
      >
        <svg
          className="c0"
          version="1.1"
          viewBox="0 0 750 472"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0,462a10,10,0,0,0,10,10H740a10,10,0,0,0,10-10V144H0Z"
          />
          <path
            d="M750,10A10,10,0,0,0,740,0H10A10,10,0,0,0,0,10V73H750Z"
          />
        </svg>
      </BaseCardIcon>
    </Default>
  </CardIconBase>
</Styled(CardIconBase)>
`;
